compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -3695,11 +3695,22 @@ private BType findAssignableType(SymbolEnv env, BLangExpression selectExp, BType
             }
         }
 
+        if (targetType.tag == TypeTags.UNION) {
+            if (((BUnionType) targetType).getMemberTypes().contains(symTable.xmlType)) {
```

Comment:
You can get ((BUnionType) targetType).getMemberTypes() out

Reply:
`findAssignableType` method has been removed in a later PR(https://github.com/ballerina-platform/ballerina-lang/pull/26195)